logging/examples/features.rs

```diff
@@ -0,0 +1,12 @@
+use holochain_logging::{prelude::*, Logger};
+
+pub fn main() {
+    let _guard = Logger::new();
+
+    trace!("logging a trace message");
+    debug!("debug values"; "x" => 1, "y" => -1);
+    info!("some interesting info"; "where" => "right here");
+    warn!("be cautious!"; "why" => "you never know...");
+    error!("wrong {}", "foobar"; "type" => "unknown");
+    crit!("abandoning test");
+}
```

Comment:
@dymayday I find the the call instantiating the logger here a bit suspect. I feel either the method `new()` is misleading and instead should be named to imply a singleton pattern or that we register a logger reference with the macro system in a more explicit manner. Your thoughts?